cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,251 @@
+			ClusterProfile: resolvedFlow.ClusterProfile,
+			Pre:            internalStepsToAPISteps(resolvedFlow.Pre),
+			Test:           internalStepsToAPISteps(resolvedFlow.Test),
+			Post:           internalStepsToAPISteps(resolvedFlow.Post),
+		}
+		newStep.MultiStageTestConfiguration = &multiStageConfig
+		result = append(result, newStep)
+	}
+	return result, nil
+}
+
+func internalStepsToAPISteps(steps []types.TestStep) []api.TestStep {
+	var newSteps []api.TestStep
+	for _, step := range steps {
+		newStep := api.TestStep{
+			LiteralTestStep: &api.LiteralTestStep{
+				As:          step.As,
+				From:        step.From,
+				Commands:    step.Commands,
+				ArtifactDir: step.ArtifactDir,
+				Resources:   step.Resources,
+			},
+		}
+		newSteps = append(newSteps, newStep)
+	}
+	return newSteps
+}
+
+func reloadRegistry(o *options) error {
+	references, chain, workflows, err := load.Registry(o.regPath)
+	if err != nil {
+		return err
+	}
+	o.resolver = registry.NewResolver(references, chain, workflows)
+	return nil
+}
+
+func populateWatcher(watcher *fsnotify.Watcher, root string) error {
+	return filepath.Walk(root, func(path string, file os.FileInfo, err error) error {
+		// We only need to watch directories as creation, deletion, and writes
+		// for files in a directory trigger events for the directory
+		if file != nil && file.IsDir() {
+			//log.Debugf("Adding %s to watch list", path)
+			err = watcher.Add(path)
+			if err != nil {
+				return fmt.Errorf("Failed to add watch on directory %s: %v", path, err)
+			}
+		}
+		return nil
+	})
+}
+
+func main() {
+	opts := options{}
+	flag.StringVar(&opts.regPath, "registry", "", "Path to step registry")
+	flag.StringVar(&opts.configPath, "config", "", "Path to config dirs")
+	logLevel := flag.String("log-level", "info", "Level at which to log output.")
+	flag.Parse()
+	level, err := log.ParseLevel(*logLevel)
+	if err != nil {
+		log.Fatalf("invalid --log-level: %v", err)
+	}
+	log.SetLevel(level)
+
+	c := coalescer{loader: sync.NewCond(&sync.Mutex{})}
+	c.reload(opts)
+
+	// watch for changes in config directories
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("Failed to create new watcher: %v", err)
+	}
+	defer watcher.Close()
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				//log.Debugf("Received %v event for %s", event.Op, event.Name)
+				if event.Op == fsnotify.Remove {
+					//log.Debugf("Removing %s from watches", event.Name)
+					watcher.Remove(event.Name)
+				}
+				// reload registry
+				go c.reload(opts)
+				// add new files to be watched
+				if err := populateWatcher(watcher, opts.regPath); err != nil {
+					log.Errorf("Failed to update fsnotify watchlist: %v", err)
+				}
+			case err := <-watcher.Errors:
+				log.Errorf("error: %v", err)
+			}
+		}
+	}()
+	populateWatcher(watcher, opts.regPath)
+
+	http.HandleFunc("/config", resolveConfig(&opts))
+	http.ListenAndServe(":8080", nil)
```

Comment:
Since we already have it imported, we might as well use the interrupts library for this https://github.com/kubernetes/test-infra/blob/master/prow/interrupts/interrupts.go#L152